resnet/main.py

```diff
@@ -0,0 +1,400 @@
+parser.add_argument('--lr', default=0.1, type=float)
+parser.add_argument('--epochs', default=200, type=int, metavar='N',
+                    help='number of total epochs to run')
+parser.add_argument('--weightDecay', default=0.0005, type=float)
+parser.add_argument('--epoch_step', default='[60,120,160]', type=str,
+                    help='json list with epochs to drop lr on')
+parser.add_argument('--lr_decay_ratio', default=0.2, type=float)
+parser.add_argument('--resume', default='', type=str)
+
+# Device options
+parser.add_argument('--save', default='/tmp/cifar', type=str,
+                    help='save parameters and logs in this folder')
+parser.add_argument('--ngpu', default=1, type=int,
+                    help='number of GPUs to use for training, 0 to for CPU')
+parser.add_argument('--gpu_id', default='0', type=str,
+                    help='id(s) for CUDA_VISIBLE_DEVICES')
+
+# TODO: move this to main
+opt = parser.parse_args()
+print 'parsed options:', vars(opt)
+
+os.environ['CUDA_VISIBLE_DEVICES'] = opt.gpu_id
+epoch_step = json.loads(opt.epoch_step)
+
+use_cuda = torch.cuda.is_available() and opt.ngpu > 0
+
+
+class RandomCrop(object):
+    """Random cropping with zero-padding or reflections
+    note: have to use legacy nn here because there's no padding with reflection
+    in PIL or skimage (could use OpenCV, but pain to install)
+    """
+    def __init__(self, pad=4, crop_type='zero'):
+        if crop_type == 'zero':
+            self.module = SpatialZeroPadding(pad, pad, pad, pad)
+        elif crop_type == 'reflection':
+            self.module = SpatialReflectionPadding(pad, pad, pad, pad)
+        self.pad = pad
+
+    def __call__(self, tensor):
+        padded = self.module.forward(tensor.unsqueeze(0))[0]
+        x = np.random.randint(0, self.pad*2)
+        y = np.random.randint(0, self.pad*2)
+        return padded[:, y:y+tensor.size(1), x:x+tensor.size(2)].contiguous()
+
+
+def get_iterator(dataset, mode):
+    """Creates iterator for train (mode=True) or test (mode=False) subset
+    """
+    mean = [125.3, 123.0, 113.9]
+    std = [63.0,  62.1,  66.7]
+
+    randomflip = lambda x: np.fliplr(x) if np.random.random() > 0.5 else x
+    normalize = lambda x: (x.astype(np.float32) - mean) / std
+    totensor = lambda x: torch.from_numpy(x.transpose(2, 0, 1)).float()
+
+    if mode:
+        transforms = [randomflip, normalize, totensor,
+                      RandomCrop(4, 'reflection')]
+    else:
+        transforms = [normalize, totensor]
+
+    dataset_constr = getattr(datasets, dataset)
+    ds = dataset_constr('.', train=mode, download=True)
+    data = ds.train_data if mode else ds.test_data
+    labels = ds.train_labels if mode else ds.test_labels
+    ds_tnt = tnt.dataset.TensorDataset([data.transpose(0, 2, 3, 1), labels])
+    ds_tnt = ds_tnt.transform({0: tnt.transform.compose(transforms)})
+    return ds_tnt.parallel(batch_size=opt.batchSize, shuffle=mode,
```

Comment:
Can't you use core data loaders in here? I can't see any functionality that's not supported (TensorDataset is also in the core).